space_packet_parser/packets.py

```diff
@@ -245,21 +256,47 @@ def __init__(self, *args, raw_data: bytes = b"", **kwargs):
     @property
     def header(self) -> dict:
         """The header content of the packet."""
+        warnings.warn("The header property is deprecated and will be removed in a future release. "
+                      "To access the header fields of a CCSDS packet, use the CCSDSPacketBytes class.")
         return dict(list(self.items())[:7])
 
     @property
     def user_data(self) -> dict:
         """The user data content of the packet."""
+        warnings.warn("The user_data property is deprecated and will be removed in a future release. "
+                      "To access the user_data fields of a CCSDS packet, use the CCSDSPacketBytes class.")
         return dict(list(self.items())[7:])
 
+class CCSDSPacket(Packet):
+    """Packet representing parsed data items from CCSDS packet(s). DEPRECATED
+
+    This class is deprecated and will be removed in a future release. Use the Packet class instead.
+    In an XTCE representation, there is no guarantee that the CCSDS packet header will be defined
+    as individual elements. If you want to access those elements, you can use the CCSDSPacketBytes
+    class to extract the header fields with specific methods.
+
+    Container that stores the raw packet data (bytes) as an instance attribute and the parsed
+    data items in a dictionary interface. A ``CCSDSPacket`` generally begins as an empty dictionary that gets
+    filled as the packet is parsed. The first 7 items in the dictionary make up the
+    packet header (accessed with ``CCSDSPacket.header``), and the rest of the items
+    make up the user data (accessed with ``CCSDSPacket.user_data``). To access the
+    raw bytes of the packet, use the ``CCSDSPacket.raw_data`` attribute.
+    """
+    def __init__(self, *args, **kwargs):
+        warnings.warn("The CCSDSPacket class is deprecated and will be removed in a future release. "
+                    "Use the Packet class instead (no CCSDS prefix).")
+        super().__init__(*args, **kwargs)
+
 
 def ccsds_generator(
             binary_data: Union[BinaryIO, socket.socket, bytes],
             *,
             buffer_read_size_bytes: Optional[int] = None,
             show_progress: bool = False,
             skip_header_bytes: int = 0,
-) -> Iterator[RawPacketData]:
+            combine_segmented_packets: bool = False,
+            secondary_header_bytes: int = 0,
+) -> Iterator[CCSDSPacketBytes]:
     """A generator that reads raw packet data from a filelike object or a socket.
 
     Each iteration of the generator yields a ``RawPacketData`` object that makes up
```

Comment:
...yields a `CCSDSPacketBytes` object...
But if we made it inherit from RawPacketData, then this would technically be true.